pkg/confgen/confgen_test.go

```diff
@@ -18,54 +18,104 @@
-## all parameters needed for the creation of the configuration
-##
+func getConfGen() *ConfGen {
+	return &ConfGen{}
+}
+
+func Test_checkHeader(t *testing.T) {
+	filename := "/tmp/header.check.txt"
+	fakeFilename := "/tmp/fake_file.does.exist"
+	wrongHeader := "#wrong_confgen"
+	cg := getConfGen()
+	err := cg.checkHeader(fakeFilename)
+	require.Error(t, err)
+
+	err = os.WriteFile(filename, []byte(wrongHeader), 0644)
+	require.Equal(t, err, nil)
+	err = cg.checkHeader(filename)
+	require.Error(t, err)
+
+	err = os.WriteFile(filename, []byte(definitionHeader), 0644)
+	require.Equal(t, err, nil)
+	err = cg.checkHeader(filename)
+	require.NoError(t, err)
+}
+
+const networkDefinitionConfiguration = `#fl2m_confgen
 description:
   test description
-ingest:
-  collector:
-    port: 8888
+details:
+  test details
+usage:
+  test usage
+labels:
+  - test
+  - label
+transform:
+  rules:
+    - input: testInput
+      output: testOutput
+      type: add_service
+      parameters: proto
+extract:
+  aggregates:
+    - name: test_aggregates
+      by:
+        - service
+      operation: sum
+      recordKey: test_record_key
 encode:
+  type: prom
   prom:
-    port: 7777
-    prefix: prefix
+    metrics:
+      - name: test_metric
+        type: gauge
+        valuekey: test_aggregates_value
+        labels:
+          - by
+          - aggregate
+visualization:
+  type: grafana
+  grafana:
+    - expr: 'test expression'
+      type: graphPanel
+      dashboard: test
+      title:
+        Test grafana title
 `
 
-func getConfGen() *ConfGen {
-	return &ConfGen{}
-}
+func Test_parseFile(t *testing.T) {
+	fakeFilename := "/tmp/fake_file.does.exist"
+	filename := "/tmp/parse_file.check.txt"
+	cg := getConfGen()
+	err := cg.parseFile(fakeFilename)
+	require.Error(t, err)
 
-func expectedConfig() *Config {
-	return &Config{
-		Description: "test description",
-		Encode: ConfigEncode{
-			Prom: api.PromEncode{
-				Port:   7777,
-				Prefix: "prefix",
-			},
-		},
-		Ingest: ConfigIngest{
-			Collector: api.IngestCollector{
-				Port: 8888,
-			},
-		},
-	}
+	err = os.WriteFile(filename, []byte(networkDefinitionConfiguration), 0644)
+	require.Equal(t, err, nil)
+	err = cg.parseFile(filename)
+	require.NoError(t, err)
 }
 
-func Test_parseConfigFile(t *testing.T) {
-	filename := "/tmp/config"
+func Test_getDefinitionFiles(t *testing.T) {
+	dirPath := "/tmp/getDefinitionFilesTest"
+	filename := "/def.yaml"
 	cg := getConfGen()
-	err := os.WriteFile(filename, []byte(testConfig), 0644)
-	require.Equal(t, err, nil)
-	config, err := cg.parseConfigFile(filename)
-	require.Equal(t, err, nil)
-	require.Equal(t, config, expectedConfig())
+	err := os.MkdirAll(dirPath, 0755)
+	require.NoError(t, err)
+	err = os.WriteFile(dirPath+filename, []byte(networkDefinitionConfiguration), 0644)
+	require.NoError(t, err)
+	files := cg.getDefinitionFiles(dirPath)
+	require.Equal(t, 1, len(files))
+	require.Equal(t, dirPath+filename, files[0])
```

Comment:
I think that something like the following is more expressive:
```go
expected := []string{path.Join(dirPath, filename)}
require.ElemetsMatch(t, expected, files)
```

Reply:
fixed